src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+import { Fact } from "@/types/conversations.types";
+import EditViewpointFact from "@/components/AuthorViewpoint/EditViewpointFact";
+import { MagnifyingGlassIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Select, Button } from "@mantine/core";
+import { useState, useEffect } from "react";
+import { allFacts } from "@/mock/conversationMock";
+
+type FactListCardProps = {
+    viewpointFactList: Fact[];
+    setViewpointFactList: (FactList: Fact[]) => void;
+};
+
+export default function FactListCard({
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts); // eslint-disable-line
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) == selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
```

Comment:
When writing mutations that reply on previous state it is not recommended to take the state variable as they do not update until rerendered, which is not triggered immideatly and instead can be batched. The current method results in race conditions and using `useEffect(prev => [...prev, new])` is recommended